gtsfm/common/image.py

```diff
@@ -48,27 +89,43 @@ def get_intrinsics_from_exif(self) -> Optional[Cal3Bundler]:
             intrinsics matrix (3x3).
         """
 
-        if self.exif_data is None or len(self.exif_data) == 0:
-            return None
+        img_w_px = self.width
+        img_h_px = self.height
+        max_size = max(img_w_px, img_h_px)
 
-        focal_length_mm = self.exif_data.get("FocalLength")
+        # Initialize focal length by `DEFAULT_FOCAL_LENGTH_FACTOR * max(width, height)`.
+        focal_length_px = DEFAULT_FOCAL_LENGTH_FACTOR * max_size
 
-        if focal_length_mm is None:
-            return None
+        # Read focal length prior from exif.
+        if self.exif_data is not None and len(self.exif_data) > 0:
 
-        try:
-            sensor_width_mm = Image.sensor_width_db.lookup(
-                self.exif_data.get("Make"),
-                self.exif_data.get("Model"),
-            )
-        except (AttributeError, LookupError):
-            # TODO(yanwei.du): Set a prior focal length, e.g. 1.5 * max(self.width, self.height).
-            return None
+            # Read from `FocalLengthIn35mmFilm`.
+            focal_length_35_mm = self.exif_data.get("FocalLengthIn35mmFilm")
+            if focal_length_35_mm is not None and focal_length_35_mm > 0:
+                focal_length_px = focal_length_35_mm / 35.0 * max_size
+            else:
+                # Read from `FocalLength` mm.
+                focal_length_mm = self.exif_data.get("FocalLength")
 
-        img_w_px = self.width
-        img_h_px = self.height
-        focal_length_px = max(img_h_px, img_w_px) * focal_length_mm / sensor_width_mm
+                # Compute sensor width, either from database or from EXIF.
+                if focal_length_mm is not None and focal_length_mm > 0:
+
+                    sensor_width_mm = 0.0
+
+                    try:
+                        sensor_width_mm = Image.sensor_width_db.lookup(
+                            self.exif_data.get("Make"),
+                            self.exif_data.get("Model"),
+                        )
+                    except (AttributeError, LookupError):
+                        sensor_width_mm = self.__compute_sensor_width_from_exif()
+
+                    if sensor_width_mm > 0.0:
+                        focal_length_px = focal_length_mm / sensor_width_mm * max_size
+
+        assert focal_length_px > 0
```

Comment:
nit: would prefer to raise rather than assert here:

```python
if focal_length <= 0:
    raise ValueError("Focal length must be positive value.")
```

Reply:
Done

Reply:
Resolved, thanks!